### PR TITLE
Fix a race condition when allocating ports.

### PR DESCRIPTION
**Pull request by obino**
When multiple application starts at the same time, depending on how long
each application takes to load, there could be a race condition when
allocating ports to the load balancer. This patch makes sure that any
ports promised to an application, is not used.

### CODE REVIEW

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/800/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/801/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/815/

File: AppController/djinn.rb [hunk 1]
```diff
@@ -665,35 +665,13 @@ def relocate_app(appid, http_port, https_port, secret)
...
-      end
-
-      if [http_port, https_port].include?(info['nginx_https'])
-        return "Error: Port in use by nginx for app #{app}"
-      end
-
-      if [http_port, https_port].include?(info['haproxy'])
-        return "Error: Port in use by haproxy for app #{app}"
-      end
+    if find_lowest_free_port(http_port, http_port, appid) == -1
```
Comment (menivaitsi) [inline AppController/djinn.rb]: Is this right?: (http_port, http_port, appid)

Comment (menivaitsi) [inline AppController/djinn.rb]: Should one of them be https_port ?

File: AppController/djinn.rb [hunk 2]
```diff
@@ -4905,15 +4889,54 @@ def find_lowest_free_port(starting_port, ending_port=0)
         break
       end
 
+      # Make sure the port is not already allocated to any application.
+      # This is important when applications starts at the same time since
```
Comment (menivaitsi) [inline AppController/djinn.rb]: starts -> start

File: AppController/djinn.rb (hunk 1 above)
Comment (obino) [inline AppController/djinn.rb]: Yes it is correct: we are not looking for a range here, but if that specific port is available.

Comment (menivaitsi) [inline AppController/djinn.rb]: Ok, can you add a comment that explains this?

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/819/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/820/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/824/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/828/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/829/

**Pull request merged**